### PR TITLE
media-radio:*: fix pkgcheck issues

### PR DESCRIPTION
**Pull request by fkobi**
---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Review by ZeroChaos- — APPROVED** [review]

**Review by dl1jbe — APPROVED** [review]

**Review by a17r — CHANGES_REQUESTED** [review]
If you are going to touch EAPI-7 ebuilds for minor stylistic changes, at least bump them to EAPI-8 at the same time (with all the necessary care doing so).

That also means you do the revbump necessary for an EAPI change and leave the existing stable ebuild alone.

File: media-radio/ax25-apps/ax25-apps-0.0.8_rc5.ebuild [hunk 1]
```diff
@@ -1,31 +1,29 @@
...
 
+BDEPEND="virtual/pkgconfig"
 RDEPEND="
 	>=dev-libs/libax25-0.0.12_rc2
 	sys-libs/ncurses:=
 	virtual/pkgconfig
 	!dev-ruby/listen
 "
 DEPEND="${RDEPEND}"
-BDEPEND="virtual/pkgconfig"
```
Comment (a17r) [inline media-radio/ax25-apps/ax25-apps-0.0.8_rc5.ebuild:26]: unnecessary change, BDEPEND comes last in `skel.ebuild` template.

File: media-radio/fsync-mdc1200-decode/fsync-mdc1200-decode-1.0-r1.ebuild [hunk 2]
```diff
@@ -12,17 +12,16 @@ if [[ ${PV} == *9999* ]]; then
...
 fi
 
 LICENSE="GPL-2+"
 SLOT="0"
-IUSE=""
 
+BDEPEND="virtual/pkgconfig"
 DEPEND="media-libs/libpulse"
 RDEPEND="${DEPEND}"
-BDEPEND="virtual/pkgconfig"
```
Comment (a17r) [inline media-radio/fsync-mdc1200-decode/fsync-mdc1200-decode-1.0-r1.ebuild:25]: unnecessary change, BDEPEND comes last in `skel.ebuild` template.

File: media-radio/fsync-mdc1200-decode/fsync-mdc1200-decode-9999.ebuild [hunk 3]
```diff
@@ -12,17 +12,16 @@ if [[ ${PV} == *9999* ]]; then
...
 fi
 
 LICENSE="GPL-2+"
 SLOT="0"
-IUSE=""
 
+BDEPEND="virtual/pkgconfig"
 DEPEND="media-libs/libpulse"
 RDEPEND="${DEPEND}"
-BDEPEND="virtual/pkgconfig"
```
Comment (a17r) [inline media-radio/fsync-mdc1200-decode/fsync-mdc1200-decode-9999.ebuild:25]: unnecessary change, BDEPEND comes last in `skel.ebuild` template.

**Review by a17r — CHANGES_REQUESTED** [review]

File: media-radio/ax25-apps/ax25-apps-0.0.8_rc5-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit autotools
+
+MY_P="${P/_/-}"
+
+DESCRIPTION="Basic AX.25 (Amateur Radio) user tools, additional daemons"
+HOMEPAGE="http://www.linux-ax25.org"
```
Comment (a17r) [inline media-radio/ax25-apps/ax25-apps-0.0.8_rc5-r1.ebuild:11]: `HOMEPAGE` is dead, is there a new one?

File: media-radio/ax25-apps/ax25-apps-0.0.8_rc5-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+SRC_URI="http://www.linux-ax25.org/pub/${PN}/${MY_P}.tar.gz"
```
Comment (a17r) [inline media-radio/ax25-apps/ax25-apps-0.0.8_rc5-r1.ebuild:12]: Equally dead, is there a new source?

File: media-radio/ax25-apps/ax25-apps-0.0.8_rc5-r1.ebuild [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+S="${WORKDIR}/${MY_P}"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="amd64 ppc x86"
```
Comment (a17r) [inline media-radio/ax25-apps/ax25-apps-0.0.8_rc5-r1.ebuild:17]: Needs to be bumped with all ~arch KEYWORDS.

File: media-radio/ax25-apps/ax25-apps-0.0.8_rc5-r1.ebuild [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+	!dev-ruby/listen
+"
+DEPEND="${RDEPEND}"
+BDEPEND="virtual/pkgconfig"
+
+src_prepare() {
+	eapply_user
+	# fix missing prototype for malloc
+	sed -i -e "/^#include /i #include <stdlib.h>" ax25ipd/routing.c || die
+	eapply "${FILESDIR}"/${PN}-0.0.8_rc5-tinfo.patch
```
Comment (a17r) [inline media-radio/ax25-apps/ax25-apps-0.0.8_rc5-r1.ebuild:32]: In general it would be better to call `eapply_user` or `default´ *after* our own changes.

```suggestion
	# fix missing prototype for malloc
	sed -i -e "/^#include /i #include <stdlib.h>" ax25ipd/routing.c || die
	eapply "${FILESDIR}"/${PN}-0.0.8_rc5-tinfo.patch
	eapply_user
```

File: media-radio/ax25-apps/ax25-apps-0.0.8_rc5-r1.ebuild [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+src_install() {
+	default
+
+	newinitd "${FILESDIR}"/ax25ipd.rc ax25ipd
+	newinitd "${FILESDIR}"/ax25mond.rc ax25mond
+	newinitd "${FILESDIR}"/ax25rtd.rc ax25rtd
+
+	rm -rf "${ED}"/usr/share/doc/ax25-apps || die
```
Comment (a17r) [inline media-radio/ax25-apps/ax25-apps-0.0.8_rc5-r1.ebuild:43]: don't we want to know if something fails here? Using `-f` will mask that.

**Review by a17r — CHANGES_REQUESTED** [review]

**Comment by a17r** [comment]
> @a17r is it okay now?

Did you test-build your EAPI-8 bump to `media-radio/ax25-apps`?

Ultimately it is for @dl1jbe to again review their ebuild.

File: media-radio/ax25-apps/ax25-apps-0.0.8_rc5-r1.ebuild [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+src_install() {
+	default
+
+	newinitd "${FILESDIR}"/ax25ipd.rc ax25ipd
+	newinitd "${FILESDIR}"/ax25mond.rc ax25mond
+	newinitd "${FILESDIR}"/ax25rtd.rc ax25rtd
+
+	rm -r "${ED}"/usr/share/doc/${PN}* || die
```
Comment (a17r) [inline media-radio/ax25-apps/ax25-apps-0.0.8_rc5-r1.ebuild:43]: That's weird, are there more than one dir? Can you show the content of `/usr/share/doc/` in the build image dir? That would be an issue by itself.

Comment (eli-schwartz) [reply media-radio/ax25-apps/ax25-apps-0.0.8_rc5-r1.ebuild:43]: As a quick hacky debug thing, you could change that to `rm -v -r` so that you see what it's removing in the build.log :)

Comment (a17r) [reply media-radio/ax25-apps/ax25-apps-0.0.8_rc5-r1.ebuild:43]: Ah, okay, that would be `${PF}` instead. https://devmanual.gentoo.org/ebuild-writing/variables/index.html

The *proper* way to fix this though would be to look into the build system and make a patch disabling the creation/installation of these doc archives in the first place.

**Review by dl1jbe — APPROVED** [review]

**Comment by dl1jbe** [comment]
Just one comment wrt SRC_URI and Homepage (Saw it after approve):

The original archive of AX-25 code and the related CVS/GIT repo was on http://www.linux-ax25.org. Now that these server is defunct it changed to https://linux-ax25.in-berlin.de.

See the README in https://github.com/ve7fet/linuxax25 which is the base for the code at packet-radio.net/ax25 you relied upon.

**Comment by dl1jbe** [comment]
> visiting https://linux-ax25.in-berlin.de gives this message:
> 
> "Sorry! This site is experiencing technical difficulties.

Right. Seems like the front end wiki is defunct but the https://linux-ax25.in-berlin.de/{pub|scm|cgit} sub directories are working fine.

> I am not sure what should I do regarding your message.

I would suggest to add both links (..in-berlin.de... and ..packet-radio.net... in HOMEPAGE and SRC_URI with the in-berlin first. That would respect the main source but give a fallback if the first one proved to be unavailable.

File: media-radio/ax25-apps/ax25-apps-0.0.8_rc5-r1.ebuild [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+S="${WORKDIR}/${MY_P}"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
+
+RDEPEND="
+	>=dev-libs/libax25-0.0.12_rc2:=
+	sys-libs/ncurses:=
+	virtual/pkgconfig
```
Comment (dl1jbe) [inline media-radio/ax25-apps/ax25-apps-0.0.8_rc5-r1.ebuild:28]: virtual/pkgconfig seems wrong in RDEPEND.

Comment (dl1jbe) [reply media-radio/ax25-apps/ax25-apps-0.0.8_rc5-r1.ebuild:28]: Otherwise looks good.

**Comment by dl1jbe** [comment]
> > Seems like the front end wiki is defunct but the [https://linux-ax25.in-berlin.de/{pub|scm|cgit}](https://linux-ax25.in-berlin.de/%7Bpub%7Cscm%7Ccgit%7D) sub directories are working fine.
> 
> scm does not work for me.

Soory, my fault. I meant .../pub/scm.

**Review by dl1jbe — APPROVED** [review]
Looks good. Thnaks for the work.
